### PR TITLE
Grayscale maps in Dark theme (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by mechaffin**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

1. Use "grayscale_dark" map style for the dark theme. Continue to use the "road" map style for the light theme.
1. Use azure maps controls for zoom (remove our custom controls for this).
1. Add azure maps style selector control. If user selects a non-default style for their current theme, that will be honored even if they change the theme. Otherwise, the style follows our theme settings.

*Note: pins on the grayscale_dark map are also grayscale... which means there is no differentiation between critical and warning pins when in this style. Will try to discuss with maps team.* 

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-remote-monitoring-webui/1102)

### CODE REVIEW

File: src/components/pages/dashboard/panels/map/azureMap.js [hunk 1]
```diff
@@ -5,19 +5,29 @@ import React, { Component } from 'react';
...
   componentWillReceiveProps(nextProps) {
-    if (!this.map && nextProps.azureMapsKey) {
-      this.initializeMap(nextProps.azureMapsKey);
+    if (!this.map && nextProps.theme && nextProps.azureMapsKey) {
+      this.initializeMap(nextProps.theme, nextProps.azureMapsKey);
+    }
+    // When the theme changes, also change the style for the map.
+    // If the user manually chose a map style different than the theme,
+    // that choice should be honored. Otherwise, use the default style for the theme.
+    if (this.map && nextProps.theme !== this.props.theme) {
```
Comment (spryor) [inline src/components/pages/dashboard/panels/map/azureMap.js]: These can be merged into a single `if`

File: src/components/pages/dashboard/panels/map/azureMap.js [hunk 2]
```diff
@@ -31,11 +41,22 @@ export class AzureMap extends Component {
...
+  initializeMap(theme, azureMapsKey) {
     this.map = new AzureMaps.Map('map', {
       'subscription-key': azureMapsKey,
       center: Config.mapCenterPosition,
-      zoom: 11
+      zoom: 11,
+      style: this.getMapStyle(theme)
+    });
+
+    var styleSelector = new AzureMaps.control.StyleControl();
```
Comment (spryor) [inline src/components/pages/dashboard/panels/map/azureMap.js]: const

Comment (mechaffin) [reply src/components/pages/dashboard/panels/map/azureMap.js]: Darn it! That's what I get for cut&Paste sample code!  Haha.

**Comment by mechaffin** [comment]
Closing the PR as the work will be done later in Vanadium and involve refactoring to use new datasource layers in Azure Maps... to give us better control over the pins and icons.

**Pull request closed without merging**